src/libostree/ostree-repo-pull.c

```diff
@@ -1369,6 +1530,15 @@ scan_commit_object (OtPullData         *pull_data,
   if (!ostree_repo_load_commit (pull_data->repo, checksum, &commit, &commitstate, error))
     goto out;
 
+  if (pull_data->summary)
```

Comment:
This `pull_data->summary` condition is unnecessary if we drop the `ensure_collection_ref_in_summary` check from above.